File: src/com/wang/Game2dEngine/painter/Painter.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.wang.Game2dEngine.painter;

import com.wang.Game2dEngine.painter.interfaces.IEngineGraphics;
import com.wang.Game2dEngine.painter.interfaces.IPainter;
import com.wang.Game2dEngine.painter.interfaces.IUserInteraction;
import com.wang.Game2dEngine.painter.interfaces.IWindow;
import java.awt.Component;
import java.awt.Dimension;
import java.awt.Graphics;
import java.awt.event.ComponentAdapter;
import java.awt.event.MouseListener;
import java.awt.image.BufferedImage;

/**
 *
 * @author ricolwang
 */
public class Painter implements IPainter, IUserInteraction, IWindow
{

    private Panel panel = new Panel();
    protected BufferedImage theImage;
    protected IEngineGraphics theEngineGraphics;

    @Override
    public int getHeight()
    {
        return panel.getHeight();
    }

    @Override
    public int getWidth()
    {
        return panel.getWidth();
    }

    @Override
    public Component getComponent()
    {
        return panel;
    }

    @Override
    public void addMouseListener(MouseListener listener)
    {
        panel.addMouseListener(listener);
    }

    @Override
    public void addComponentListener(ComponentAdapter componentAdapter)
    {
        panel.addComponentListener(componentAdapter);
    }

    @Override
    public void setSize(Dimension d)
    {
        panel.setSize(d);
    }

    public void render()
    {
        panel.render();
    }

    public Graphics getRenderGraphics()
    {
        return panel.getRenderGraphics();
    }

    public void painterSizeDidChanged()
    {

    }
}

File: src/com/wang/Game2dEngine/painter/Painter.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.wang.Game2dEngine.painter

import com.wang.Game2dEngine.painter.interfaces.IEngineGraphics
import com.wang.Game2dEngine.painter.interfaces.IPainter
import com.wang.Game2dEngine.painter.interfaces.IUserInteraction
import com.wang.Game2dEngine.painter.interfaces.IWindow
import java.awt.Component
import java.awt.Dimension
import java.awt.Graphics
import java.awt.event.ComponentAdapter
import java.awt.event.MouseListener
import java.awt.image.BufferedImage

/**
 * @author ricolwang
 */
open class Painter : IPainter, IUserInteraction, IWindow
{
    private val panel = Panel()
    protected var theImage: BufferedImage? = null
    protected var theEngineGraphics: IEngineGraphics? = null
    override val height: Int
        get() = panel.height
    override val width: Int
        get() = panel.width
    override val component: Component
        get() = panel

    override fun addMouseListener(listener: MouseListener?)
    {
        panel.addMouseListener(listener)
    }

    override fun addComponentListener(componentAdapter: ComponentAdapter?)
    {
        panel.addComponentListener(componentAdapter)
    }

    override fun setSize(d: Dimension?)
    {
        panel.size = d
    }

    fun render()
    {
        panel.render()
    }

    val renderGraphics: Graphics?
        get() = panel.renderGraphics

    open fun painterSizeDidChanged()
    {
    }
}